plugin/manager_linux.go

```diff
@@ -183,8 +183,8 @@ func (pm *Manager) upgradePlugin(p *v2.Plugin, configDigest digest.Digest, blobs
 				return
 			}
 
-			if err := os.Rename(backup, orig); err != nil {
-				err = errors.Wrap(err, "error restoring old plugin root on upgrade failure")
```

Comment:
This changes the `nil` value of `err` to this new error.
So effectively this failed `Rename` will return an error back to the user.